### PR TITLE
chore: upgrade webpack

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

upgrade webpack

**Did you add tests for your changes?**
yes

**If relevant, did you update the documentation?**
NA

**Summary**
Fix CI and upgrades webpack.

**Does this PR introduce a breaking change?**
no

**Other information**
Fix https://github.com/webpack/webpack-cli/pull/2728

### CODE REVIEW

**Review by snitin315 — DISMISSED** [review]

File: test/build/core-flags/module-flags.test.js [hunk 1]
```diff
@@ -114,6 +114,52 @@ describe("module config related flag", () => {
                             "myLoader",
                         ]);
                         expect(normalizeStdout(stdout)).toContain(`${propName}: 'pre'`);
+                    } else if (name.includes("compiler")) {
+                        let { stdout } = await run(__dirname, [`--${name}`, "test-compiler"]);
```
Comment (jamesgeorge007) [inline test/build/core-flags/module-flags.test.js:118]: Why not using `const`? There're multiple instancess in the test suite.

Comment (alexander-akait) [reply test/build/core-flags/module-flags.test.js:118]: We need rule in eslint config to enforce const

Comment (anshumanv) [reply test/build/core-flags/module-flags.test.js:118]: yep should be const

Comment (anshumanv) [reply test/build/core-flags/module-flags.test.js:118]: will add

Comment (alexander-akait) [reply test/build/core-flags/module-flags.test.js:118]: Let's fix and merge

**Review by alexander-akait — APPROVED** [review]

**Comment by anshumanv** [comment]
@webpack/cli-team can anyone approve? need merge

**Review by snitin315 — APPROVED** [review]

**Pull request merged**